content/document.js

```diff
@@ -262,7 +262,7 @@ function findByURL(url, ...args) {
   const [bareURL, hash = ""] = url.split("#", 2);
   const doc = read(urlToFolderPath(bareURL), ...args);
   if (doc && hash) {
-    doc.url = `${doc.url}#${hash}`;
+    return { url: `${doc.url}#${hash}`, ...doc };
```

Comment:
I don't know why but I don't understand this. If you want to NOT mutate but to create a copied object, but with a different `url` key, then normally you do:
```javascript
const foo = { url: "something", other: "stuff" };
console.log({ url: "new", ...foo });
```
will output:
```
{ url: 'something', other: 'stuff' }
```

But I appreciate the underlying core problem. You get a memorized thing back from a memorizing function, but then you mutate it which is just wrong. So why would your PR fix the problem??

Reply:
Bah! I figured it out. My comment was correct. You're creating a new object the wrong way. Here's how I reproduced the problem:
```javascript
const { findByURL } = require("./content/document.js");

console.log(
  findByURL("/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords").url
);
console.log(
  findByURL("/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#other").url
);
console.log(
  findByURL("/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords").url
);
```
And when run with `NODE_ENV=production node dummy.js` it spits out:
```
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords#other
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords#other#keywords
```
which looks exactly like the problem we have. 
If I change to your solution the output becomes:
```
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar
```
But if you create the new object clone with `url` last, the output (correctly) becomes:
```
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#other
/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords
```